File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/false_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class FalseCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(FalseCC.conditions(), condition)

    @staticmethod
    def conditions() -> Set[Condition]:
        return {Condition.FALSE}

File: golang_vm/uPIMulator/src/device/linker/kernel/instruction/cc/false_cc.go
package cc

import (
	"errors"
)

type FalseCc struct {
	condition Condition
}

func (this *FalseCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		FALSE: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *FalseCc) Condition() Condition {
	return this.condition
}
